pkg/controller/volume/attachdetach/statusupdater/node_status_updater.go

```diff
@@ -87,7 +89,16 @@ func (nsu *nodeStatusUpdater) UpdateNodeStatuses() error {
 				"Could not update node status for %q; re-marking for update. %v",
 				nodeName,
 				err)
-
+			failedNodeNumber := nodeNumber
+			var i int
+			//re-marking remaining nodes for update
+			for nodeName, _ = range nodesToUpdate {
+				i++
+				if i < failedNodeNumber+1 {
+					continue
+				}
+				nsu.actualStateOfWorld.SetNodeStatusUpdateNeeded(nodeName)
```

Comment:
Should I add log statement for these nodes also similar to [this](https://github.com/kubernetes/kubernetes/pull/107976/files#diff-4505a6c5e6e8d09f1f5ff559a9ca14e4efa62131d211ea151f2f86dc295ca982R70-R73) statement which was added for the failed node .

Reply:
This logic might be dangerous. I don't think there is any guarantee in the go language that iterating over a map twice will result in the same iteration order (see for instance https://go.dev/blog/maps - Iteration order)
So if we want to  proceed with aborting the call altogether if there is an error in that clause, we might want to introduce an abort boolean in the main scope and set it to true if there is such an error and have a check at the beginning of the loop that will check for abort and if true will call nsu.actualStateOfWorld.SetNodeStatusUpdateNeeded and continue to the next entry

Reply:
@jfremy When we will change the value of `var abourt bool` inside the condition `if err := nsu.updateNodeStatus(nodeName, nodeObj, attachedVolumes); err != nil {)` then how will be able to use it to re-mark the remaining nodes because at the end of this condition we `return err` which will end the `function`, so how can we check the value of `abort` for the remaining nodes for re-marking them.

Reply:
The return err needs to be removed in that scenario

Reply:
So there is no need to return this error `err := nsu.updateNodeStatus(nodeName, nodeObj, attachedVolumes); err != nil` in the function.

Reply:
The beginning of the loop would have the abort condition
```
if abort {
  nsu.actualStateOfWorld.SetNodeStatusUpdateNeeded(nodeName)
  continue
}
```

while the error path for updateNodeStatus would look like this:
```
if err := nsu.updateNodeStatus(nodeName, nodeObj, attachedVolumes); err != nil {
    // If update node status fails, reset flag statusUpdateNeeded back to true
    // to indicate this node status needs to be updated again
    nsu.actualStateOfWorld.SetNodeStatusUpdateNeeded(nodeName)
    klog.V(2).Infof(
        "Could not update node status for %q; re-marking for update. %v",
        nodeName,
        err)`
    abort = true
    continue
}
```
with abort being defined as a bool in the outer scope of the function

But again, that's only if we want to go in the abort the whole call direction. I don't have the context of why the original code writer made that choice. Maybe replacing `return err` with `continue` would be ok too

Reply:
Done, thanks for the help

Reply:
abort looks not very necessary here? Basically it needs iterate the whole node list, and update node status. If any node status update failed, mark nodeStatusUpdateNeed in actual state of world, and continue to the next item on the list. The return error in this function is not very necessary either, it is right now just for logging.

Reply:
There are different possible node update failures. 
1. In case of somehow node does not exist, node update failed. In this case, we want to continue updating other nodes.
2. In case API server issue, all update might fail. In this case, abort the rest operation is a good choice to reduce loading api more.
But I think to be safer, we should try to continue to update the rest of the nodes in the list. Otherwise, one node update failure will always block the rest of the list.  

Basically it iterates the whole node list, and update node status. If any node status update failed, mark nodeStatusUpdateNeed in actual state of world, and continue to the next item on the list. The return error in this function is not very necessary either.